AspNetCoreServerSide/Views/Home/Index.cshtml

```diff
@@ -63,6 +87,208 @@
+                    {
+                        method: 'POST',
+                        cache: 'no-cache',
+                        body: new URLSearchParams(new FormData(document.querySelector('#frmCreate')))
+                    })
+                    .then((response) => {
+                        table.ajax.reload();
+                        $('#createModal').modal('hide');
+                        document.querySelector('#frmCreate').reset();
+                    })
+                    .catch((error) => {
+                        console.log(error);
+                    });
+            });
+
+        $(document)
+            .off('click', '.btnEdit')
+            .on('click', '.btnEdit', function () {
+                const id = $(this).attr('data-key');
+
+                fetch(`/Home/Edit/${id}`,
+                    {
+                        method: 'GET',
+                        cache: 'no-cache'
+                    })
+                    .then((response) => {
+                        return response.text();
+                    })
+                    .then((result) => {
+                        $('#editPartial').html(result);
+                        $('#editModal').modal('show');
+                    })
+                    .catch((error) => {
+                        console.log(error);
+                    });
+            });
+
+        $(document)
+            .off('click', '#btnUpdate')
+            .on('click', '#btnUpdate', function () {
+                fetch('/Home/Edit/',
+                    {
+                        method: 'PUT',
+                        cache: 'no-cache',
+                        body: new URLSearchParams(new FormData(document.querySelector('#frmEdit')))
+                    })
+                    .then((response) => {
+                        table.ajax.reload();
+                        $('#editModal').modal('hide');
+                        $('#editPartial').html('');
+                    })
+                    .catch((error) => {
+                        console.log(error);
+                    });
+            });
+
+        $(document)
+            .off('click', '.btnDelete')
+            .on('click', '.btnDelete', function () {
+                const id = $(this).attr('data-key');
+
+                if (confirm('Are you sure?')) {
+                    fetch(`/Home/Delete/${id}`,
+                        {
+                            method: 'DELETE',
+                            cache: 'no-cache'
+                        })
+                        .then((response) => {
+                            table.ajax.reload();
+                        })
+                        .catch((error) => {
+                            console.log(error);
+                        });
+                }
+            });
+
+        $('#btnExternalSearch').click(function () {
+            table.column('0:visible').search($('#txtExternalSearch').val()).draw();
+        });
+    }
+    </script>
```

Comment:
Please place the scripts inside `app.js`. Don't delete app.js file. As a best practice always keep scripts in separate file. This can be handled as shown below.

In the `Index.cshtml`,
```html
<script src="~/lib/datatables/js/dataTables.buttons.min.js" asp-append-version="true"></script>
<script type="text/javascript">
    let langFileUrl = '@langFileUrl';
    let exportToExcelText = '@Localizer["Export to Excel"]';
    let createText = '@Localizer["Create"]';
</script>
<script src="~/js/app.js" asp-append-version="true"></script>
```

Then in `app.js` use the above variables,
```js
language: {
        url: langFileUrl,
},
buttons: [
        {
            text: exportToExcelText,
        },
        {
            text: createText,
        }
]
```